### DISCUSSION TITLE
[Bug]: Hyprlock config file does not work as intended

### DISCUSSION THREAD

**Discussion by chrollorifat** [prasanthrangan/hyprdots#1388] [General]
### Description

Editing the hyprlock.conf in the ~/.config/hypr does not really work. Even after following proper syntax according to manual, multiple error pops up after sourcing the hyprlock.conf in hyprland.conf. It would be very helpful if we could provide a customized lockscreen in our rice instead of the default, bare minimum hyprlock.

**Comment by dieBakterie** [comment]
Do you source they layout.conf at the bottom If yes irt overwrites what you Changes in the conf file

**Reply by chrollorifat** [reply]
No, I don't source the layout.conf.

**Reply by dieBakterie** [reply]
Okay First of you don't need to source the hyprlock in hyprland. Second do you have Switched Out swaylock for hyprlock in `keybindings.conf`?

**Reply by chrollorifat** [reply]
ooh. Thank you so much. I didn't notice we were using swaylock.

**Comment by dieBakterie** [comment]
> ## Bug Report
> 
> ### Description
> 
> Editing the hyprlock.conf in the ~/.config/hypr does not really work. Even after following proper syntax according to manual, multiple error pops up after sourcing the hyprlock.conf in hyprland.conf. It would be very helpful if we could provide a customized lockscreen in our rice instead of the default, bare minimum hyprlock.
> 
> 

A Screenshot with those Errors would Help

**Comment by kRHYME7** [comment]
@ChrolloRifat  Hyprlock? We did not implement it yet. 

> pops up after sourcing the hyprlock.conf in hyprland.conf

Also Why source the hyprlock configuration to hyprland.conf? 

I move this to discussions as this is too unrelated for now.

**Reply by kRHYME7** [reply]
> instead of the default, bare minimum hyprlock. 

Installer also do not have any hyprlock mentioned
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/b373e256-0c0e-4a84-9526-23731c66980d)

**Reply by chrollorifat** [reply]
Will it work if we don't source the hyprlock config to the hyprland.conf??

**Reply by kRHYME7** [reply]
BTW where did you got the hyprlock.conf? 

> Will it work if we don't source the hyprlock config to the hyprland.conf??

There's no such thing as sourcing hyprlock.conf to hyprland.conf. Both are different

If no ~/.config/hypr/hyprlock.conf exu=isted then it will be just a transparent lockscreen.

**Reply by kRHYME7** [reply]
` hyprlock -c /path/to/your/config   ` 

this is how to indicate what config file to use

or defaulted to ` ~/.config/hypr/hyprlock `

**Reply by kRHYME7** [reply]
will add a template soon in the PR

**Reply by chrollorifat** [reply]
ooh, thanks, buddy. Got it. Will we be getting a customized lockscreen in the future? or should i work on one and create a pull a request?

**Reply by kRHYME7** [reply]
https://github.com/prasanthrangan/hyprdots/pull/1173

We will have that, but the scaling is not good for now. 
It is possible to implement it with all sizes set to scaling of 1 but that would lead to more issues being flooded to this repo. Specially more user got fractional scaling which hyprlock.conf handles it differently 

For now  you guys can discuss on how hyprlock should work in this [PR ](https://github.com/prasanthrangan/hyprdots/pull/1173)
